rand.c

```diff
@@ -35,22 +35,13 @@
 // You are supposed to replace the random8() and random32() function with your own secure code.
 // There is also a possibility to replace the random_buffer() function as it is defined as a weak symbol.
 
-static uint8_t random8(void)
+uint32_t random32(void)
 {
-    // Linear congruential generator used in glibc
+	// Linear congruential generator used in glibc
     // https://en.wikipedia.org/wiki/Linear_congruential_generator
     static int seed = 0;
-    seed = (1103515245 * seed + 12345) & 0x7FFFFFFF;
-    return seed & 0xFF;
-}
-
-uint32_t random32(void)
-{
-    uint32_t r1 = random8();
-    uint32_t r2 = random8();
-    uint32_t r3 = random8();
-    uint32_t r4 = random8();
-    return ((r1 << 24) | (r2 << 16) | (r3 << 8) | r4);
+    seed = (1664525 * seed + 1013904223) & 0xFFFFFFFF;
```

Comment:
we don't need `& 0xFFFFFFFF;` now

Reply:
Thanks to the uint32 type? Got it

Reply:
It's not needed because we use all 32 bits, so `& 0xFFFFFFFF` is essentially a no-op for a 32-bit integral type. (I assume it would work for signed 32 bit int too).